In [1]:
import pandas as pd


In [3]:
hate_crime = pd.read_csv('CSV/Hate_Crimes.csv')

In [4]:
hate_crime.describe()

Year   Anti-Male  Anti-Female  Anti-Transgender  \
count   362.000000  362.000000   362.000000        362.000000   
mean   2012.367403    0.008287     0.016575          0.005525   
std       1.692525    0.090782     0.127848          0.074226   
min    2010.000000    0.000000     0.000000          0.000000   
25%    2011.000000    0.000000     0.000000          0.000000   
50%    2012.000000    0.000000     0.000000          0.000000   
75%    2014.000000    0.000000     0.000000          0.000000   
max    2015.000000    1.000000     1.000000          1.000000   

       Anti-Gender Non-Conforming   Anti-Age*  Anti-White  Anti-Black  \
count                  362.000000  362.000000  362.000000  362.000000   
mean                     0.052486    0.046961    0.325967    1.850829   
std                      0.246879    0.499866    1.030741    2.678500   
min                      0.000000    0.000000    0.000000    0.000000   
25%                      0.000000    0.000000    0.000000    0.000000   
50%                      0.000000    0.000000    0.000000    1.000000   
75%                      0.000000    0.000000    0.000000    2.000000   
max                      2.000000    9.000000   11.000000   18.000000   

       Anti-American Indian/Alaskan Native  Anti-Asian       ...         \
count                           362.000000  362.000000       ...          
mean                              0.008287    0.176796       ...          
std                               0.090782    0.675419       ...          
min                               0.000000    0.000000       ...          
25%                               0.000000    0.000000       ...          
50%                               0.000000    0.000000       ...          
75%                               0.000000    0.000000       ...          
max                               1.000000    8.000000       ...          

       Anti-Male Homosexual  Anti-Female Homosexual  Anti-LGBT (Mixed Group)  \
count            362.000000              362.000000               362.000000   
mean               1.513812                0.232044                 0.118785   
std                4.361895                0.726446                 0.414061   
min                0.000000                0.000000                 0.000000   
25%                0.000000                0.000000                 0.000000   
50%                0.000000                0.000000                 0.000000   
75%                1.000000                0.000000                 0.000000   
max               36.000000                7.000000                 4.000000   

       Anti-Heterosexual  Anti-Bisexual  Anti-Physical Disability  \
count         362.000000     362.000000                 362.00000   
mean            0.002762       0.005525                   0.01105   
std             0.052559       0.074226                   0.10468   
min             0.000000       0.000000                   0.00000   
25%             0.000000       0.000000                   0.00000   
50%             0.000000       0.000000                   0.00000   
75%             0.000000       0.000000                   0.00000   
max             1.000000       1.000000                   1.00000   

       Anti-Mental Disability  Total Incidents  Total Victims  Total Offenders  
count               362.00000       362.000000     362.000000       362.000000  
mean                  0.01105        10.140884      10.563536        12.005525  
std                   0.10468        16.915614      17.613348        20.735238  
min                   0.00000         1.000000       1.000000         1.000000  
25%                   0.00000         1.000000       1.000000         1.000000  
50%                   0.00000         3.000000       3.000000         3.000000  
75%                   0.00000        10.750000      11.000000        12.000000  
max                   1.00000       101.000000     106.000000       113.000000  

[8 rows x 42 columns]

In [5]:
hate_crime.shape

(362, 44)

In [6]:
hate_crime.columns

Index(['County', 'Year', 'Crime Type', 'Anti-Male', 'Anti-Female',
       'Anti-Transgender', 'Anti-Gender Non-Conforming', 'Anti-Age*',
       'Anti-White', 'Anti-Black', 'Anti-American Indian/Alaskan Native',
       'Anti-Asian', 'Anti-Native Hawaiian/Pacific Islander',
       'Anti-Multi-Racial Groups', 'Anti-Other Race', 'Anti-Jewish',
       'Anti-Catholic', 'Anti-Protestant', 'Anti-Islamic (Muslim)',
       'Anti-Multi-Religious Groups', 'Anti-Atheism/Agnosticism',
       'Anti-Religious Practice Generally', 'Anti-Other Religion',
       'Anti-Buddhist', 'Anti-Eastern Orthodox (Greek, Russian, etc.)',
       'Anti-Hindu', 'Anti-Jehovahs Witness', 'Anti-Mormon',
       'Anti-Other Christian', 'Anti-Sikh', 'Anti-Hispanic', 'Anti-Arab',
       'Anti-Other Ethnicity/National Origin', 'Anti-Non-Hispanic*',
       'Anti-Male Homosexual', 'Anti-Female Homosexual',
       'Anti-LGBT (Mixed Group)', 'Anti-Heterosexual', 'Anti-Bisexual',
       'Anti-Physical Disability', 'Anti-Mental Disa

In [8]:
hate_crime['Crime Type'].value_counts()

Crimes Against Persons    204
Property Crimes           158
Name: Crime Type, dtype: int64

In [15]:
import matplotlib.pyplot as plt
%matplotlib inline

In [20]:
hate_crime.index

RangeIndex(start=0, stop=362, step=1)

In [30]:
hate_crime_subset = hate_crime[['Year','Anti-Black']]

In [49]:
hate_crime_subset.sort(['Anti-Black'], ascending=[False]).head(20)

C:\Users\Rittick\Anaconda3\lib\site-packages\ipykernel\__main__.py:1: FutureWarning: sort(columns=....) is deprecated, use sort_values(by=.....)
  if __name__ == '__main__':


Year  Anti-Black
85   2012          18
81   2010          18
300  2012          18
302  2013          16
304  2014          13
83   2011          10
110  2010          10
149  2011           9
150  2012           9
0    2010           9
355  2011           9
299  2012           8
89   2014           8
356  2012           8
154  2014           8
301  2013           7
295  2010           7
87   2013           7
162  2012           7
155  2014           7

In [42]:
hate_crime_subset_2 = hate_crime[['Year','Anti-Asian']]

In [46]:
hate_crime_subset_2.sort(['Anti-Asian'], ascending=[False]).head(10)

C:\Users\Rittick\Anaconda3\lib\site-packages\ipykernel\__main__.py:1: FutureWarning: sort(columns=....) is deprecated, use sort_values(by=.....)
  if __name__ == '__main__':


Year  Anti-Asian
158  2010           8
168  2015           6
85   2012           4
160  2011           2
120  2015           2
116  2013           2
82   2010           2
231  2013           2
166  2014           2
164  2013           2

In [52]:
hate_crime_subset_3 = hate_crime[['Year','Anti-White']]

In [55]:
hate_crime_subset_3.sort(['Anti-White'], ascending=[False]).head(7)

C:\Users\Rittick\Anaconda3\lib\site-packages\ipykernel\__main__.py:1: FutureWarning: sort(columns=....) is deprecated, use sort_values(by=.....)
  if __name__ == '__main__':


Year  Anti-White
81   2010          11
85   2012           9
87   2013           6
166  2014           5
164  2013           4
116  2013           4
110  2010           3

In [57]:
hate_crime_subset_3 = hate_crime[['Year','Anti-Islamic (Muslim)']]

In [59]:
hate_crime_subset_3.sort(['Anti-Islamic (Muslim)'], ascending=False).head(10)

C:\Users\Rittick\Anaconda3\lib\site-packages\ipykernel\__main__.py:1: FutureWarning: sort(columns=....) is deprecated, use sort_values(by=.....)
  if __name__ == '__main__':


Year  Anti-Islamic (Muslim)
118  2014                     10
110  2010                      7
229  2012                      7
120  2015                      7
112  2011                      6
158  2010                      6
160  2011                      5
164  2013                      5
119  2014                      4
162  2012                      4

In [68]:
hate_crime_year = hate_crime[['Year', 'Total Incidents',
       'Total Victims', 'Total Offenders']]

In [70]:
hate_crime_year.sort(['Total Victims'], ascending=[False]).head(20)

C:\Users\Rittick\Anaconda3\lib\site-packages\ipykernel\__main__.py:1: FutureWarning: sort(columns=....) is deprecated, use sort_values(by=.....)
  if __name__ == '__main__':


Year  Total Incidents  Total Victims  Total Offenders
115  2012              101            106              104
110  2010               81             82              101
300  2012               82             82               82
120  2015               78             81              104
164  2013               72             80              113
116  2013               68             76              102
118  2014               72             76               97
158  2010               75             75               91
304  2014               75             75               84
168  2015               67             73               86
302  2013               68             68               68
119  2014               62             63               63
114  2012               56             62               87
162  2012               54             61               76
112  2011               53             61               80
166  2014               55             60               78
117  2013               55             56               57
111  2010               54             54               55
151  2012               53             53               53
113  2011               53             53               60

What happened in 2012 that resulted in 106 victims? What were the events and who were responsible?

Check out this article and search for '2012'. The article confirms the peak in such incidents. http://www.nbcnews.com/storyline/charleston-church-shooting/hate-crime-america-numbers-n81521